src/main/kotlin/no/nav/syfo/tilgang/TilgangApi.kt

```diff
@@ -4,23 +4,56 @@ import io.ktor.http.*
 import io.ktor.server.application.*
 import io.ktor.server.response.*
 import io.ktor.server.routing.*
+import no.nav.syfo.application.api.auth.isMissingNAVIdent
 import no.nav.syfo.util.getBearerHeader
 import no.nav.syfo.util.getCallId
 
 const val tilgangApiBasePath = "/api/tilgang"
+const val enhetNr = "enhetNr"
 
 fun Route.registerTilgangApi(
     tilgangService: TilgangService,
 ) {
     route(tilgangApiBasePath) {
         get("/navident/syfo") {
+            val callId = call.getCallId()
             val token = call.getBearerHeader()
                 ?: throw IllegalArgumentException("Failed to check syfo tilgang for veileder. No Authorization header supplied")
+            if (token.isMissingNAVIdent()) {
+                throw IllegalArgumentException("Failed to check enhetstilgang for veileder. No NAV ident in token")
+            }
+
+            val tilgang = tilgangService.hasTilgangTilSyfo(
+                token = token,
+                callId = callId,
+            )
+
+            if (tilgang.harTilgang) {
+                call.respond(tilgang)
+            } else {
+                call.respond(
+                    status = HttpStatusCode.Forbidden,
+                    message = tilgang
+                )
+            }
+        }
+
+        get("/navident/enhet/{$enhetNr}") {
             val callId = call.getCallId()
+            val token = call.getBearerHeader()
+                ?: throw IllegalArgumentException("Failed to check enhetstilgang for veileder. No Authorization header supplied")
+            if (token.isMissingNAVIdent()) {
+                throw IllegalArgumentException("Failed to check enhetstilgang for veileder. No NAV ident in token")
+            }
+
+            val enhetNr = call.parameters[enhetNr]
+                ?: throw IllegalArgumentException("No EnhetNr found in path param")
+            val enhet = Enhet(enhetNr)
```

Comment:
Jeg vet ikke helt hva som må/burde eksplisitt kaste exception her, for at det skal være lett å skjønne hva som er lov og ikke 🤷 I noen apper har vi vel kanskje dette i en try/catch?

Nå er det noen ting som eksplisitt kaster exception: manglende token, manglende NAV-ident og manglende enhet header.
Konstruktøren til Enhet vil også kaste exception hvis enheten ikke er gyldig, men det ser man jo ikke her nå.

Alle exceptions vil fanges i statusPages og returnere 400/500 til den som gjorde kallet.

Reply:
Jeg har ingen sterke formeninger, bortsett fra at jeg ikke har noe problemer med hvordan det står her nå 😊